src/global_mapper/global_mapper.cc

```diff
@@ -92,6 +92,51 @@ void GlobalMapper::InsertPointCloud(const PointCloud::ConstPtr& cloud_ptr) {
   }
 }
 
+void GlobalMapper::FlattenMap() {
+  int ixyz[3] = {0};
+  double xyz[3] = {0.0};
+  double xy[2] = {0.0};
+  float occ = 0.0;
+  float occ_temp = 0.0;
+  for (int i=0; i < voxel_map_ptr_->dimensions[0]; ++i) {
+    ixyz[0] = i;
+
+    for (int j=0; j < voxel_map_ptr_->dimensions[1]; ++j) {
+      ixyz[1] = j;
+
+      // reset mean
+      occ = 0.0;
+      occ_temp = 0.0;
+
+      for (int k=0; k < voxel_map_ptr_->dimensions[2]; ++k) {
+        ixyz[2] = k;
+
+        // get coordinates from voxel_map
+        voxel_map_ptr_->tableToWorld(ixyz, xyz);
+
+        // check bounds
+        if (xyz[2] < params_.pixel_min_z_abs_ ||
+            xyz[2] > params_.pixel_max_z_abs_) {
+          continue;
+        }
+
+        // printf("z: %0.2f", xyz[2]);
+
+        // get max over z
+        occ_temp = voxel_map_ptr_->readValue(ixyz);
+        if (occ_temp > occ) {
+          occ = occ_temp;
+        }
+      }
+
+      // insert into pixel map by position
+      xy[0] = xyz[0];
+      xy[1] = xyz[1];
+      pixel_map_ptr_->writeValue(xy, occ);
+    }
+  }
+}
+
 std::unique_lock<std::mutex> GlobalMapper::CloudLock() {
   return std::unique_lock<std::mutex>(cloud_mutex_);
```

Comment:
Still feel like unique_locks are strictly better and that lock_guard is completely superseded by it.  Added the accessor functions to get references to mutexes instead of unique_locks, though.  272477481969a4f16cf2569a3048c2ec1d72fee4.